### PR TITLE
#84: Add flight status feedback

### PR DESCRIPTION
**Pull request by rpeterman-gp**
#84 

### To Review
- [ ] Checkout branch.
- [ ] Run `docker-compose build`.
- [ ] Run `docker-compose up`.
- [ ] Start http-proxy container.
- [ ] Go to [grove.prx.docker](https://grove.prx.docker).
- [ ] Note changes to campaign status column.
- [ ] Inspect a Flight Name cell, and add `error` to its `class` attribute.
- [ ] Note row shows a red indictor and highlights in red.
- [ ] Go to a flight.
- [ ] Add some thing to the title.
- [ ] Note valid change state is reflected by tab indicator.
- [ ] Remove value from required field.
- [ ] Note invalid change state is reflected by tab indicator.
- [ ] Change a date.
- [ ] Note error is displayed between goal form and availability table in red.
- [ ] Note error state is reflected by tab indicator and error icon is shown.
- [ ] Go to another flight by clicking another tab.
- [ ] Note previous modified flight state is still reflected by tab indicator.
- [ ] Refresh page to discard changes and reset state.
- [ ] Make a valid change to the flight.
- [ ] Save campaign.
- [ ] Note flight tab state indicator returns to the selected state.

### CODE REVIEW

File: src/app/campaign/availability/availability.component.ts [hunk 1]
```diff
@@ -117,6 +126,23 @@ export class AvailabilityComponent {
     );
   }
 
+  getFlightErrors() {
```
Comment (sandikbarr) [inline src/app/campaign/availability/availability.component.ts]: suggestion: slight refactor to get function and in the template using `ngIf` with `as`:
```
<ul class="errors" *ngIf="errors as flightErrors">
        <li class="error" *ngFor="let error of flightErrors">
```
```
get errors() {
    const errors = [];

    if (this.allocationPreviewError) {
      errors.push(`Got error ${this.allocationPreviewError.status} from allocation preview.`);
    }

    if (this.flight.status_message) {
      errors.push(this.flight.status_message);
    }

    return errors;
  }
```

File: src/app/campaign/campaign.component.scss [hunk 2]
```diff
@@ -1,36 +1,70 @@
...
   padding-top: 0;
-}
 
-.mat-list-base .mat-list-item {
-  background: #ddd;
-  color: #0089bd;
-  font-weight: 700;
-  margin: 0 0 2px;
-}
+  .mat-list-item {
```
Comment (sandikbarr) [inline src/app/campaign/campaign.component.scss:17]: this looks good 👍

File: src/sass/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,23 @@
+@import '~@angular/material/theming';
+@import 'colors';
+
+//// Colors
```
Comment (sandikbarr) [inline src/sass/_variables.scss:4]: 👍 thanks

**Review by sandikbarr — APPROVED** [review]
slight refactor suggestion on the errors in the availability component, otherwise looks good. thanks for the material updates also

File: src/app/campaign/availability/availability.component.ts (hunk 1 above)
Comment (rpeterman-gp) [reply src/app/campaign/availability/availability.component.ts]: This looks much cleaner. I always seem to forget about setters/getters. I pushed up a change with this suggestion. Thank you.

Comment (sandikbarr) [reply src/app/campaign/availability/availability.component.ts]: Looks great, thanks!

**Review by sandikbarr — APPROVED** [review]
looks good!

**Pull request merged**